evaluation/agent_bench/run_infer.py

```diff
@@ -0,0 +1,436 @@
+        'instruction': instruction,
+        'metadata': metadata,
+        'history': histories,
+        'error': state.error if state and state.error else None,
+        'test_result': {
+            'agent_answer': agent_answer,
+            'final_answer': final_ans,
+            'check_method': comparison_method,
+            'result': test_result,
+        },
+    }
+
+    # clean up
+    if os.path.exists(instance_workspace):
+        shutil.rmtree(instance_workspace)
+    # Close the sandbox
+    try:
+        sandbox.close()
+    except docker.errors.NotFound as e:
+        logger.error(f'Failed to close sandbox: {e}')
+    return output
+
+
+def find_subdirectories(directory):
+    return [
+        name
+        for name in os.listdir(directory)
+        if os.path.isdir(os.path.join(directory, name))
+    ]
+
+
+def load_datasets_from_local(key: str, _dir: str) -> [dict]:
+    _datasets = {}
+    if os.path.exists(_dir):
+        for filename in os.listdir(_dir):
+            if filename.endswith('.json'):  # assuming the datasets are in csv format
+                file_path = os.path.join(_dir, filename)
+                with open(file_path, 'r') as r:
+                    _datasets[filename] = json.load(r)
+    return flatten(key, _datasets)
+
+
+def flatten(key: str, _data: dict) -> [dict]:
+    flat_data = []
+    for subkey, values in _data.items():
+        if isinstance(values, list):
+            for i in range(len(values)):
+                value = values[i]
+                value['instance_id'] = f'{key}_{subkey}_{i}'
+                value['task_idx'] = key
+                flat_data.append(value)
+    return flat_data
+
+
+if __name__ == '__main__':
+    # =============================================
+    # load datasets
+    # =============================================
+
+    dataset = load_dataset('iFurySt/AgentBench')
+    agent_bench_tests = dataset['osbench'].to_pandas()
+    logger.info(f'Loaded {len(agent_bench_tests)} tests.')
+
+    # =============================================
+    # handle arguments and prepare for evaluation
+    # =============================================
+
+    if args.llm_config:
+        specified_llm_config = get_llm_config_arg(args.llm_config)
+        if specified_llm_config:
+            config.llm = specified_llm_config
+    logger.info(f'Config for evaluation: {config}')
+
+    # TEST METADATA
+    agent_cls = args.agent_cls
+    assert (
+        agent_cls in AGENT_CLS_TO_FAKE_USER_RESPONSE_FN
+    ), f'Unsupported agent class: {agent_cls}'
+    model_name = config.llm.model.split('/')[-1]
+    max_iterations = args.max_iterations
+    eval_note = ''
+    if args.eval_note is not None:
+        eval_note += '_N_' + args.eval_note
+    eval_op_dir = str(
+        os.path.join(
+            args.eval_output_dir,
+            'agent_bench',
+            agent_cls,
+            model_name + '_maxiter_' + str(max_iterations) + eval_note,
```

Comment:
```suggestion
            model_name + '_maxiter_' + str(max_iterations) + eval_note,
            'osbench',
```
Maybe this can help separate the log for each task. But this maybe not needed at the moment.